server/endpoints/Review.ts

```diff
@@ -0,0 +1,59 @@
+import { body } from "express-validator";
+import { getCrossListOR } from "common/CourseCard";
+import { Endpoint } from "../endpoints";
+import { Reviews, Classes } from "../dbDefs";
+
+interface CourseId {
+  courseId: string;
+}
+
+/**
+ * Get a course with this course_id from the Classes collection
+ */
+export const getCourseById: Endpoint<CourseId> = {
+  guard: [body("courseId").notEmpty().isAscii()],
+  callback: async (courseId: CourseId) => {
+    try {
+      // check: make sure course id is valid and non-malicious
+      const regex = new RegExp(/^(?=.*[A-Z0-9])/i);
+      if (regex.test(courseId.courseId)) {
+        return await Classes.findOne({ _id: courseId.courseId }).exec();
+      }
+      return { error: "Malformed Query" };
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log("Error: at 'getCourseById' method");
+      // eslint-disable-next-line no-console
+      console.log(error);
+      return { error: "Internal Server Error" };
+    }
+  },
+};
+
+/**
+ * Get list of review objects for given class from class _id
+ */
+export const getReviewsByCourseId: Endpoint<CourseId> = {
+  guard: [body("courseId").notEmpty().isAscii()],
+  callback: async (courseId: CourseId) => {
+    try {
+      const regex = new RegExp(/^(?=.*[A-Z])/i);
+      if (regex.test(courseId.courseId)) {
+        const course = await getCourseById.callback(courseId);
```

Comment:
Sorry for the late comment but could the common functionality between this endpoint and getCourseById be factored out into a function called by both? Thinking through future endpoints, they'll probably need to get courses by ids as well. While we can simply call the endpoint again like is done here, it isn't best practice: https://softwareengineering.stackexchange.com/questions/345406/is-it-bad-design-to-internally-call-api-endpoints-from-within-the-api-instance.

Reply:
Yea that makes sense. I can remove getCourseById from the endpoints just have that be a utility function. Should I make a utils.ts and put it there (since its not an endpoint it doesn't make sense to keep it in Review.ts)? Or did you have something else in mind?

Reply:
I think you can keep the getCourseById endpoint because the front end uses the getCourseById Meteor method in a few places and put a function that actually gets the course in that utils file you suggested. Then the getCourseById and any other endpoints that need to get course info can call that function

Reply:
Ah, I see what you mean. I think I misunderstood the first time.